tests/unit/aiplatform/test_datasets.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import pytest
+
+from unittest import mock
+from importlib import reload
+from unittest.mock import patch
+
+from google.api_core import operation
+from google.auth import credentials as auth_credentials
+
+from google.cloud import aiplatform as aip
+from google.cloud.aiplatform import Dataset
+from google.cloud.aiplatform import initializer
+
+from google.cloud.aiplatform_v1beta1 import GcsSource
+from google.cloud.aiplatform_v1beta1 import GcsDestination
+from google.cloud.aiplatform_v1beta1 import ImportDataConfig
+from google.cloud.aiplatform_v1beta1 import ExportDataConfig
+from google.cloud.aiplatform_v1beta1 import DatasetServiceClient
+from google.cloud.aiplatform_v1beta1 import Dataset as GapicDataset
+
+_TEST_PROJECT = "test-project"
+_TEST_LOCATION = "us-central1"
+_TEST_ALT_LOCATION = "europe-west4"
+_TEST_ID = "1028944691210842416"
+_TEST_PARENT = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}"
+_TEST_NAME = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/datasets/{_TEST_ID}"
+_TEST_ALT_NAME = (
+    f"projects/{_TEST_PROJECT}/locations/{_TEST_ALT_LOCATION}/datasets/{_TEST_ID}"
+)
+
+_TEST_INVALID_LOCATION = "us-central2"
+_TEST_INVALID_NAME = f"prj/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/{_TEST_ID}"
+
+_TEST_LABEL = {"team": "experimentation", "trial_id": "x435"}
+_TEST_DISPLAY_NAME = "my_dataset_1234"
+_TEST_METADATA_SCHEMA_URI = "gs://my-bucket/schema-9876.yaml"
+
+_TEST_IMPORT_SCHEMA_URI = "gs://google-cloud-aiplatform/schemas/1.0.0.yaml"
+_TEST_SOURCE_URI = "gs://my-bucket/my_index_file.jsonl"
+_TEST_SOURCE_URIS = [
+    "gs://my-bucket/index_file_1.jsonl",
+    "gs://my-bucket/index_file_2.jsonl",
+    "gs://my-bucket/index_file_3.jsonl",
+]
+_TEST_INVALID_SOURCE_URIS = ["gs://my-bucket/index_file_1.jsonl", 123]
+_TEST_DATA_LABEL_ITEMS = {}
+
+_TEST_OUTPUT_DIR = "gs://my-output-bucket"
+
+
+class TestDataset:
+    def setup_method(self):
+        reload(initializer)
+        reload(aip)
+
+    @pytest.fixture
+    def get_dataset_mock(self):
+        with patch.object(DatasetServiceClient, "get_dataset") as get_dataset_mock:
+            get_dataset_mock.return_value = GapicDataset(
```

Comment:
this can become difficult to maintain to make sure the mocks return correct responses.  we should look into replays.